src/error/mod.rs

```diff
@@ -114,65 +146,21 @@ impl fmt::Display for NetavarkError {
             NetavarkError::Sysctl(e) => write!(f, "Sysctl error: {}", e),
             NetavarkError::Serde(e) => write!(f, "JSON Decoding error: {}", e),
             NetavarkError::Netlink(e) => write!(f, "Netlink error: {}", e),
-        }
-    }
-}
-
-impl PartialEq for NetavarkError {
```

Comment:
I read the commit message and I don't know if I agree. Checking exact equality, including message, seems useful?

Reply:
Why keep unused code that is completely useless. We really should not check errors with `==` with it turn does string compare. I already check for specific errors and that results in `match {}` statements.

Reply:
Also it is in fact impossible to compare error like that, it would require us to actual have an pre-existing error which we could compare to. AFAICS no rust library provides error in such a way, instead they use enum matching as described here: https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html